docs/migration/wsc55/icons.md

```diff
@@ -0,0 +1,140 @@
+The new version features not only many hundreds of new icons but also focused a lot more on icon consistency, namely the proper alignment of icons within the grid.
+
+The previous implementation of Font Awesome 4 included shims for Font Awesome 3 that was used before, the most notable one being the `.icon` notation instead of `.fa` as seen in Font Awesome 4 and later.
+In addition, Font Awesome 5 introduced the concept of different font weights to separate icons which was further extended in Font Awesome 6.
+
+In WoltLab Suite 6.0 we have made the decision to make a clean cut and drop support for the Font Awesome 3 shim as well as a Font Awesome 4 shim in order to dramatically reduce the CSS size and to clean up the implementation.
+Brand icons had been moved to a separate font in Font Awesome 5, but since more and more fonts are being added we have stepped back from relying on that font.
+We have instead made the decision to embed brand icons using inline SVGs which are much more efficient when you only need a handful of brand icons instead of loading a 100kB+ font just for a few icons.
+
+## Misuse of Icons as Buttons
+
+One pattern that could be found every here and then was the use of icons as buttons.
+Using icons in buttons is fine, as long as there is a readable title and that they are properly marked as buttons.
+
+A common misuse looks like this:
+
+```smarty
+<span class="icon icon16 fa-times pointer jsMyDeleteButton" data-some-object-id="123"></span>
+```
+
+This example has a few problems, for starters it is not marked as a button which would require both `role="button"` and `tabindex="0"` to be recognized as such.
+Additionally there is no title which leaves users clueless about what the option does, especially visually impaired users are possibly unable to identify the icon.
+
+WoltLab Suite 6.0 addresses this issue by removing all default styling from `<button>` elements, making them the ideal choice for button type elements.
+
+```smarty
+<button class="jsMyDeleteButton" data-some-object-id="123" title="descriptive title here">{icon name='xmark'}</button>
+```
+
+The icon will appear just as before, but the button is now properly accessible.
+
+## Using CSS Classes With Icons
+
+It is strongly discouraged to apply CSS classes to icons themselves.
+Icons inherit the text color from the surrounding context which removes the need to manually apply the color.
+
+If you ever need to alter the icons, such as applying a special color or transformation, you should wrap the icon in an element like `<span>` and apply the changes to that element instead.
+
+## Using Icons in Templates
+
+The new template function `{icon}` was added to take care of generating the HTML code for icons, including the embedded SVGs for brand icons.
+Icons in HTML should not be constructed using the actual HTML element, but instead always use `{icon}`.
+
+```smarty
+<button class="button">{icon name='bell'} I‘m a button with a bell icon</button>
```

Comment:
```suggestion
<button class="button">{icon name='bell'} I’m a button with a bell icon</button>
```

The correct apostrophe is “thick at the top, thin at the bottom” (matching the closing quote).

Reply:
Good catch. I sometimes miss the shift modifier and the difference between `‘` and `’` in my editor is unnoticable…